### PR TITLE
Verify a storageproof obtained with getProof

### PR DESCRIPTION
**Pull request by ewynx**
This is a small working example of verifying a storageProof obtained with getProof, as described in https://eips.ethereum.org/EIPS/eip-1186. It can be tested with tests in Solidity, have not yet tried it on testnet. 

## Description of functionality 
Current working (for easy testing):
- send id & json string containing key, proof elements and root (storageHash) to StorageProofChecker contract. The id is for testing purposes, to easily retrieve the result later
- StorageProofChecker sends this to the Rust program, where the data is read out and put into a trie. The storage value for the key is obtained
- Result is sent back to contract and stored there
- In tests (`StorageProofChecker.t.sol`) then the result is retrieved from the contract and compared to the value that was expected. (This verification will ultimately all be put in the Rust program)

## Tests

Testdata obtained from Optimism Goerli network. Description of how to get this and then obtain the needed json string is added in `tests/TEST_INFO.md`. 

2 specific testcases have been added. These are obtained from different contracts, blocks and indices. 

## TODOs
This has been created using the Bonsai Foundry Starter Template. The `lib` folder contain submodules as can be seen here https://github.com/risc0/bonsai-foundry-template/tree/main/lib but they have not been correctly added to this repo. It's a TODO.

There are a few first other TODOs that I didn't have the chance for:
- changing the Rust program name
- add some error handling in Rust program
- how does it work with license? There was a HC license and the starter template also had a license
- add a separate lib with functionality instead of everything directly in the Rust program (but that didn't work for me directly, so have to figure out how to make it work)

### CODE REVIEW

File: methods/guest/src/bin/fibonacci.rs
Comment (Mikerah) [inline methods/guest/src/bin/fibonacci.rs:1]: Can you rename this file so that it's clearer what it does?

**Review by Mikerah — APPROVED** [review]

File: methods/guest/src/bin/fibonacci.rs
Comment (man2706kum) [reply methods/guest/src/bin/fibonacci.rs:1]: Done

**Review by Mikerah — APPROVED** [review]

**Comment by ewynx** [comment]
@Mikerah added a fix for the submodules and there's still an open question from the initial comment

> how does it work with license? There was a HC license and the starter template also had a license

**Comment by Mikerah** [comment]
@ewynx Just use the Bonsai template license.

**Pull request merged**